turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -109,6 +110,8 @@ func (r *RemoteBlockReader) CanonicalHash(ctx context.Context, tx kv.Getter, blo
 	return rawdb.ReadCanonicalHash(tx, blockHeight)
 }
 
+var _ services.FullBlockReader = &RemoteBlockReader{}
```

Comment:
this is unnecessary in my opinion, the code wont compile without it

Reply:
I disagree.  It is there as a static type assertion to aid the development process.

What it does it cause a fast compile error - which makes it a lit easier to mange interface/implementation changes.

It has no runtime impact - but improves dev productivity.  It is also a guard against runtime type cast errors.

I'm not sure why I used it here - I suspect because I wanted to get a complete list of the missing methods.

Reply:
This feels like a Java style coding pattern which is not necessary in Go in my opinion. If you try to compile Erigon using `make erigon` or your IDE you would get a compilation error even without this static type assertion. In Goland IDE at least when I navigate to the compilation error for this it gives me a shortcut to implement missing methods.

Reply:
We using such pattern in many places. This is useful also for CI: some interfaces may be generated from .proto, some objects may become incompatible after bad-merge-conflict-resolve.